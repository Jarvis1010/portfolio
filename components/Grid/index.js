import styled from "styled-components";
import {
  spacing4,
  spacing8,
  spacing16,
  spacing32,
  spacing64,
  spacing128
} from "../constants/styled-constants";

const gaps = {
  "0": "0",
  "4": spacing4,
  "8": spacing8,
  "16": spacing16,
  "24": "1.5rem",
  "32": spacing32,
  "64": spacing64,
  "128": spacing128,
  default: spacing16
};

const createColumns = ({ columns, minWidth }) =>
  `repeat(${columns ? columns : "auto-fit"}, 
   minmax(${minWidth ? minWidth : "1px"},1fr))`;

const Grid = styled.div`
  display: grid;
  grid-template-columns: ${createColumns};
  grid-gap: ${props => gaps[props.gapSize] || gaps.default};
  width: 100%;
`;

export default Grid;
