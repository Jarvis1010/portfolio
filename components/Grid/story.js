import React, { Fragment } from "react";
import { storiesOf } from "@storybook/react";
import Grid from "./index";

const Div = ({ children }) => (
  <div style={{ border: "1px solid black" }}>{children}</div>
);

storiesOf("Grid", module)
  .add("Grid defaults to auto columns", () => (
    <Grid>
      <Div>1</Div>
      <Div>2</Div>
      <Div>3</Div>
      <Div>4</Div>
      <Div>5</Div>
      <Div>6</Div>
    </Grid>
  ))
  .add("Grid set to 3 columns", () => (
    <Grid columns={3}>
      <Div>1</Div>
      <Div>2</Div>
      <Div>3</Div>
      <Div>4</Div>
      <Div>5</Div>
      <Div>6</Div>
    </Grid>
  ))
  .add("Grid auto minWidth set to 300px", () => (
    <Fragment>
      <Div>Adjust window width</Div>
      <Grid minWidth={"300px"}>
        <Div>1</Div>
        <Div>2</Div>
        <Div>3</Div>
        <Div>4</Div>
        <Div>5</Div>
        <Div>6</Div>
      </Grid>
    </Fragment>
  ))
  .add("Grid set to 4 columns, gapSize 128", () => (
    <Fragment>
      <Div>Adjust window width</Div>
      <Grid columns={4} gapSize="128">
        <Div>1</Div>
        <Div>2</Div>
        <Div>3</Div>
        <Div>4</Div>
        <Div>5</Div>
        <Div>6</Div>
      </Grid>
    </Fragment>
  ));
