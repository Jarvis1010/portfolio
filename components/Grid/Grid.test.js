import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Grid from "./index";

describe("Grid Tests", () => {
  it("renders with proper gaps with default columns and width", () => {
    ["0", "4", "8", "16", "24", "32", "64", "128", null].forEach(gapSize => {
      const wrapper = shallow(<Grid gapSize={gapSize} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  it("renders Grid with columns passed as props", () => {
    const wrapper = shallow(<Grid columns={4} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders Grid with minWidth passed as props", () => {
    const wrapper = shallow(<Grid minWidth={100} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
