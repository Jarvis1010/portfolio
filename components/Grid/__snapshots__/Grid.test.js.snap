// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Tests renders Grid with columns passed as props 1`] = `
<StyledComponent
  columns={4}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Grid Tests renders Grid with minWidth passed as props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  minWidth={100}
/>
`;

exports[`Grid Tests renders with proper gaps with default columns and width 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="0"
/>
`;

exports[`Grid Tests renders with proper gaps with default columns and width 2`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="4"
/>
`;

exports[`Grid Tests renders with proper gaps with default columns and width 3`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="8"
/>
`;

exports[`Grid Tests renders with proper gaps with default columns and width 4`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="16"
/>
`;

exports[`Grid Tests renders with proper gaps with default columns and width 5`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="24"
/>
`;

exports[`Grid Tests renders with proper gaps with default columns and width 6`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="32"
/>
`;

exports[`Grid Tests renders with proper gaps with default columns and width 7`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="64"
/>
`;

exports[`Grid Tests renders with proper gaps with default columns and width 8`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="128"
/>
`;

exports[`Grid Tests renders with proper gaps with default columns and width 9`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-template-columns: ",
          [Function],
          ";
  grid-gap: ",
          [Function],
          ";
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize={null}
/>
`;
